xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/ConversionMatrix.h

```diff
@@ -98,27 +150,139 @@ class PrimaryToRGB : public PrimaryToXYZ
 
 //------------------------------------------------------------------------------
 
+using Matrix4 = CMatrix<4>;
+using Matrix3 = CMatrix<3>;
+using Matrix3x1 = std::array<float, 3>;
+
+/**
+ * @brief Helper class used for YUV to RGB conversions. This class can
+ *        take into account different source/destination primaries and
+ *        various other parameters.
+ *
+ */
 class CConvertMatrix
 {
 public:
   CConvertMatrix() = default;
   virtual ~CConvertMatrix() = default;
 
-  void SetColParams(AVColorSpace colSpace, int bits, bool limited, int textuteBits);
-  void SetColPrimaries(AVColorPrimaries dst, AVColorPrimaries src);
-  void SetParams(float contrast, float black, bool limited);
-  void GetYuvMat(float (&mat)[4][4]);
-  bool GetPrimMat(float (&mat)[3][3]);
+  /**
+   * @brief Set the source color space.
+   *
+   * @param colorSpace
+   * @return CConvertMatrix&
```

Comment:
Do we really need this? IMO it gives 0 information and just blows up the file. AFAIK Doxygen is smart enough to figure out the return type anyway

Reply:
Should I remove the `param` and the `return`? I'm not overly familiar with how doxygen works and I'm just using the vscode auto doxygen generator.

Reply:
IMO yes, I would remove all annotations that do not contain any additional comment/text. Doxygen automatically deduces those. They get put in the template so you can more easily fill in the text, but you can as well remove them if they are not filled.